File: Back-end/src/main/java/KJX/KJX/entity/Scrim.java
package KJX.KJX.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "scrims", indexes = {
    @Index(name = "idx_scrims_status", columnList = "status"),
    @Index(name = "idx_scrims_team_id", columnList = "team_id"),
    @Index(name = "idx_scrims_date", columnList = "date")
})
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Scrim {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "team_id", nullable = false)
    private Team team;
    
    @Column(nullable = false, length = 255)
    private String opponent;
    
    @Column(columnDefinition = "NVARCHAR(MAX)")
    private String description;
    
    @Column(nullable = false)
    private LocalDateTime date;
    
    @Column(length = 50)
    @Builder.Default
    private String status = "PENDING"; // "PENDING", "APPROVED", "REJECTED"
    
    @Column(length = 100)
    private String game; // "lol", "valorant"
    
    @Column(columnDefinition = "NVARCHAR(MAX)")
    private String notes;
    
    @Column(name = "created_at", updatable = false)
    @Builder.Default
    private LocalDateTime createdAt = LocalDateTime.now();
    
    @Column(name = "updated_at")
    @Builder.Default
    private LocalDateTime updatedAt = LocalDateTime.now();
    
    @PrePersist
    protected void onCreate() {
        this.createdAt = LocalDateTime.now();
        this.updatedAt = LocalDateTime.now();
    }
    
    @PreUpdate
    protected void onUpdate() {
        this.updatedAt = LocalDateTime.now();
    }
}

File: nodejs-backend/src/models/Scrim.js
/**
 * Scrim Model
 * Converted from Spring Boot JPA Entity
 */

const { DataTypes } = require('sequelize');

module.exports = (sequelize) => {
  const Scrim = sequelize.define('Scrim', {
    id: {
      type: DataTypes.BIGINT,
      primaryKey: true,
      autoIncrement: true
    },
    team_id: {
      type: DataTypes.BIGINT,
      allowNull: false,
      field: 'team_id',
      references: {
        model: 'teams',
        key: 'id'
      }
    },
    opponent: {
      type: DataTypes.STRING(255),
      allowNull: false
    },
    description: {
      type: DataTypes.TEXT
    },
    date: {
      type: DataTypes.DATE,
      allowNull: false
    },
    status: {
      type: DataTypes.STRING(50),
      defaultValue: 'PENDING',
      validate: {
        isIn: [['PENDING', 'APPROVED', 'REJECTED']]
      }
    },
    game: {
      type: DataTypes.STRING(100)
    },
    notes: {
      type: DataTypes.TEXT
    },
    created_at: {
      type: DataTypes.DATE,
      defaultValue: DataTypes.NOW,
      field: 'created_at'
    },
    updated_at: {
      type: DataTypes.DATE,
      defaultValue: DataTypes.NOW,
      field: 'updated_at'
    }
  }, {
    tableName: 'scrims',
    timestamps: true,
    createdAt: 'created_at',
    updatedAt: 'updated_at',
    indexes: [
      {
        fields: ['status'],
        name: 'idx_scrims_status'
      },
      {
        fields: ['team_id'],
        name: 'idx_scrims_team_id'
      },
      {
        fields: ['date'],
        name: 'idx_scrims_date'
      }
    ]
  });

  return Scrim;
};
